lightningd/channel_control.c

```diff
@@ -2271,6 +2364,103 @@ static struct command_result *json_splice_signed(struct command *cmd,
 	return result;
 }
 
+static struct command_result *json_stfu_channels(struct command *cmd,
+						 const char *buffer,
+						 const jsmntok_t *obj UNNEEDED,
+						 const jsmntok_t *params)
+{
+	struct channel *channel, **channels;
+	struct channel_id **channel_ids;
+	const jsmntok_t *channel_ids_tok, *channel_id;
+	struct command_result *result;
+	struct splice_command *cc;
+	size_t i;
+
+	if (!param_check(cmd, buffer, params,
+			 p_opt("channel_ids", param_array, &channel_ids_tok),
+			 NULL))
+		return command_param_failed();
+
+	channels = tal_arr(cmd, struct channel*, 0);
```

Comment:
The two places we load the channels (`stfu` and `abort`) have slightly different needs. The `stfu` one needs to check if the channel has an active splice command and error out, while `abort` doesn't need or want this check.

I think this reduces the value of cutting it into its own function since the channels will need to be iterated anyway.